clientv3/integration/watch_test.go

```diff
@@ -343,16 +343,45 @@ func putAndWatch(t *testing.T, wctx *watchctx, key, val string) {
 	}
 }
 
-// TestWatchResumeComapcted checks that the watcher gracefully closes in case
+func TestWatchResumeCompacted(t *testing.T) {
+	defer testutil.AfterTest(t)
+	testWatchResumeCompacted(t, nil, nil)
+}
+
+func TestWatchResumeCompactedFrequentDisconnect(t *testing.T) {
```

Comment:
is it possible to develop a dedicated test for the case we fix?

i feel `TestWatchResumeCompacted` is already hard enough to reason about (it is not deterministic. we have to sub case to care about internally)